File: fm-domain/src/main/java/io/zeitwert/dddrive/ddd/adapter/api/jsonapi/AggregateDtoAdapter.java
package io.zeitwert.dddrive.ddd.adapter.api.jsonapi;

import io.dddrive.ddd.model.Aggregate;

public interface AggregateDtoAdapter<A extends Aggregate, D extends AggregateDto<A>> {

	void toAggregate(D dto, A aggregate);

	D fromAggregate(A aggregate);

}

File: fm-domain/src/main/java/io/zeitwert/dddrive/ddd/adapter/api/jsonapi/AggregateDtoAdapter.kt
package io.zeitwert.dddrive.ddd.adapter.api.jsonapi

import dddrive.ddd.core.model.Aggregate

interface AggregateDtoAdapter<A : Aggregate, D : AggregateDto> {

	fun toAggregate(
		dto: D,
		aggregate: A,
	)

	fun fromAggregate(aggregate: A): D

}
